syntax = "proto3";

package cosim;

service CoSim {

    rpc RMVal(SimRequest) returns (SimResponse) {}

}

message SimRequest {
    int32  id = 1;
    string  type = 2;
    repeated string channels = 3;
    repeated int32 buffer = 4;
}

message SimResponse {
    int32  id = 1;
    string  type = 2;
    bool  valid = 3;
    repeated int32 buffer = 4;
}
